packetbeat/protos/cassandra/cassandra.go

```diff
@@ -0,0 +1,200 @@
+package cassandra
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/elastic/beats/packetbeat/protos"
+	"github.com/elastic/beats/packetbeat/protos/tcp"
+	"github.com/elastic/beats/packetbeat/publish"
+)
+
+// cassandra application level protocol analyzer plugin
+type cassandra struct {
+	ports        protos.PortsConfig
+	parserConfig parserConfig
+	transConfig  transactionConfig
+	pub          transPub
+}
+
+// Application Layer tcp stream data to be stored on tcp connection context.
+type connection struct {
+	streams [2]*stream
+	trans   transactions
+}
+
+// Uni-directioal tcp stream state for parsing messages.
+type stream struct {
+	parser parser
+}
+
+var (
+	debugf = logp.MakeDebug("cassandra")
+
+	// use isDebug/isDetailed to guard debugf/detailedf to minimize allocations
+	// (garbage collection) when debug log is disabled.
+	isDebug = false
+)
+
+func init() {
+	protos.Register("cassandra", New)
+}
+
+// New create and initializes a new cassandra protocol analyzer instance.
+func New(
+	testMode bool,
+	results publish.Transactions,
+	cfg *common.Config,
+) (protos.Plugin, error) {
+	p := &cassandra{}
+	config := defaultConfig
+	if !testMode {
+		if err := cfg.Unpack(&config); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := p.init(results, &config); err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+func (cassandra *cassandra) init(results publish.Transactions, config *cassandraConfig) error {
+	if err := cassandra.setFromConfig(config); err != nil {
+		return err
+	}
+	cassandra.pub.results = results
+
+	isDebug = logp.IsDebug("http")
+	return nil
+}
+
+func (cassandra *cassandra) setFromConfig(config *cassandraConfig) error {
+
+	// set module configuration
+	if err := cassandra.ports.Set(config.Ports); err != nil {
+		return err
+	}
+
+	// set parser configuration
+	parser := &cassandra.parserConfig
+	parser.maxBytes = tcp.TCP_MAX_DATA_IN_STREAM
+
+	// set parser's compressor, only `snappy` supported right now
+	if config.Compressor == Snappy {
+		parser.compressor = SnappyCompressor{}
+	} else {
+		parser.compressor = nil
+	}
```

Comment:
@urso one bit in header info of the network protocol indicates that whether it compressed or not, and the compressor should be configed during the start time